### PR TITLE
[webgpu] AddN operator support

### PR DESCRIPTION
**Pull request by haoyunfeix**
FEATURE

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4652)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@qjia7 @axinging @xhcao PTAL

File: tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  workGroupSize: [number, number, number] = [64, 1, 1];
+
+  constructor(shapes: number[][]) {
+    this.outputShape = shapes[0];
+    this.variableNames = shapes.map((_, i) => `T${i}`);
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize,
+        [this.workPerThread, 1, 1]);
+    this.shaderKey = `addN`;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts:41]: nit:  use `'addN'`

**Review by qjia7 — APPROVED** [review]

File: tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const type = getCoordsDataType(this.outputShape.length);
+    const userCode = `
+      void main() {
+        int index = int(gl_GlobalInvocationID.x);
+        for (int i = 0; i < ${this.workPerThread}; i++) {
+          int flatIndex = index * ${this.workPerThread} + i;
+          if (flatIndex < ${size}) {
+            ${type} coords = getCoordsFromFlatIndex(flatIndex);
+            ${snippets.join('\n        ')}
+            setOutput(flatIndex,${operation});
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts:67]: nit: `setOutput(flatIndex, ${operation});`. Add a space after `,`.

**Review by haoyunfeix — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4652)* status: :shipit: complete! 1 of 1 approvals obtained

___
*[tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts, line 41 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4652#-MSzwPsmHITXE9xEGx7d-r1-41:-MT-OUaACTpTvHAgA7s7:b-896fix) ([raw file](https://github.com/tensorflow/tfjs/blob/0f753956195533840d2616a9089b80a7b84cf557/tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts#L41)):*
<details><summary><i>Previously, qjia7 (Jiajia Qin) wrote…</i></summary><blockquote>

nit:  use `'addN'`
</blockquote></details>

Done.
___
*[tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4652#-MSzwPsmHITXE9xEGx7d-r1-67:-MT-OXKBDf63dG69L-FO:bb6un1j) ([raw file](https://github.com/tensorflow/tfjs/blob/0f753956195533840d2616a9089b80a7b84cf557/tfjs-backend-webgpu/src/kernels/addn_packed_webgpu.ts#L67)):*
<details><summary><i>Previously, qjia7 (Jiajia Qin) wrote…</i></summary><blockquote>

nit: `setOutput(flatIndex, ${operation});`. Add a space after `,`.
</blockquote></details>

Done

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Comment by qjia7** [comment]
@lina128 @kainino0x take a look, thanks.

**Review by kainino0x — APPROVED** [review]

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4652#-:-MT3trH58oK23RfYDiwg:bb6ywmx)* status: :shipit: complete! 3 of 1 approvals obtained

**Pull request merged**